### ISSUE TITLE
[BUG] Radio buttons dont' work correctly in a Popup when displayed a second time.

### ISSUE THREAD

**Issue by Hooterr** [CommunityToolkit/Maui#1429]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [X] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

If you add some radio buttons to a popup they work but only the first time you show the popup. Any subsequent showing of the same popup will result in semi broken behavior. Namely, the radio button needs to be clicked twice for it to take any effect.

The solution seems to be to set the `Parent` property of the `Content` of the poup to `null` when the popup is closed.
Haven't verified it but I suspect it has something to do with the last condition in this `if` statement in `RadioButton` source
https://github.com/dotnet/maui/blob/4a5000b64cb79afc20ea8308b01303b2950fd156/src/Controls/src/Core/RadioButton.cs#L444C140-L444C140

[f1.webm](https://github.com/CommunityToolkit/Maui/assets/28758721/a306789e-a1b8-45c4-99a1-e67f33d7e272)

[f2.webm](https://github.com/CommunityToolkit/Maui/assets/28758721/9e91f485-af44-457b-9087-df377da2dd6b)

### Expected Behavior

Radio buttons should not need two clicks to change state.

### Steps To Reproduce

1. Open the popup twice and try to change the radio group value.

### Link to public reproduction project repository

https://github.com/Hooterr/PopupLayoutIssue

### Environment

```markdown
- .NET MAUI CommunityToolkit: 5.3.0
- OS: macos
- .NET MAUI: 7.0.92
```

### Anything else?

WORKAROUND

```csharp
protected override Task OnClosed(object result, bool wasDismissedByTappingOutsideOfPopup)
{
    Content.Parent = null;
    return base.OnClosed(result, wasDismissedByTappingOutsideOfPopup);
}
```

**Comment by cat0363** [comment]
This issue is resolved by PR #1398.

[src\CommunityToolkit.Maui\Views\Popup\Popup.shared.cs]

    protected virtual async Task OnClosed(object? result, bool wasDismissedByTappingOutsideOfPopup)
    {
        ((IPopup)this).OnClosed(result);

        await popupDismissedTaskCompletionSource.Task;
        Parent = null;		// <= here

        dismissWeakEventManager.HandleEvent(this, new PopupClosedEventArgs(result, wasDismissedByTappingOutsideOfPopup), nameof(Closed));
    }

Therefore, it does not occur in version 6.0.0.

**Comment by pictos** [comment]
Since @cat0363 has verified, and this works on the latest version, I'll close this issue. @Hooterr feel free to reopen or open a new issue if you still face this bug
